### PR TITLE
Create resource.Default() with required attributes/default values

### LINKED ISSUE #1391: Install default SDK resources if none are configured
From a comment here: https://github.com/open-telemetry/opentelemetry-go/pull/1373#discussion_r534466997

The problem is that every exporter has convenience "installers" that do not currently allow configuring resources. The recently added `sdk/resource/resource.New()` method has a default behavior that should be used when there are no resources; this is required by the OTel spec. I think it would be best if we did not add this support to every one of the installers. 

Instead, we could have a `sync.Once()` and a default resource computed the first time it is needed. The call to `resource.New()` is not free, so we shouldn't call it more than once, but I'd say that any SDK configuration that passes `nil` as the `*Resource` should use the default value. This means we won't have to add new `With()` methods anywhere, just modify the trace and metric SDKs to get the default resource if there is none given.

### PR DESCRIPTION
**Pull request by evantorrie**
As per changes to the OpenTelemetry spec https://github.com/open-telemetry/opentelemetry-specification/pull/1294, this PR makes available a "default" Resource. If a metric or trace provider does not explicitly have a Resource assigned to it using `WithResource()` when creating the metric controller or the trace provider, the default Resource is used instead.

Resolves #1391

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

**Review by XSAM — APPROVED** [review]
LGTM

File: sdk/resource/resource.go [hunk 1]
```diff
@@ -29,7 +31,13 @@ type Resource struct {
 	labels label.Set
 }
 
-var emptyResource Resource
+var (
+	emptyResource Resource
+
+	defaultResource *Resource = func(r *Resource, err error) *Resource { return r }(
+		Detect(context.Background(), defaultServiceNameDetector{}, TelemetrySDK{}),
```
Comment (MrAlias) [inline sdk/resource/resource.go:38]: Should we surface the returned error from `Detect` with the `ErrorHandler`?

**Review by MrAlias — APPROVED** [review]
Looks good :+1:, one question about the error reporting.

File: sdk/resource/resource.go (hunk 1 above)
Comment (evantorrie) [reply sdk/resource/resource.go:38]: We can.

In the current implementation of `defaultServiceNameDetector` and `TelemetrySDK`, II don't think it's possible to return a non-nil error, but for future-proofing, it's probably good to call `otel.Handle(err)`

**Pull request merged**